paasta_tools/tron_tools.py

```diff
@@ -231,6 +233,16 @@ def get_spark_paasta_pool(self):
     def get_env(self):
         env = super().get_env()
         spark_env = {}
+        if "aws_credentials_yaml" in self.config_dict:
+            access_key, secret_key = get_aws_credentials(
```

Comment:
Can you investigate how brittle this will be when a user typo's the yaml file location?

Sometimes it is "ok" to crash in the context of a users' cli command like paasta spark-run, where it would not be ok to crash inside a lower-level piece of infrastructure.

I'll save you a click:
```
.tox/py36-linux/bin/python
Python 3.6.0 (default, Jan 13 2017, 20:56:47)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from paasta_tools.spark_tools import get_aws_credentials
/nail/home/kwa/paasta/paasta_tools/utils.py:1459: UserWarning: clog is unavailable
  warnings.warn("clog is unavailable")
>>> get_aws_credentials()
('AKIAXWG3UY5BE7JX6K5D', 'sKKMsAdVKhWB4kxXiaRIxYKqU7rHFTX7X0/89g1T')
>>> get_aws_credentials(aws_credentials_yaml='bla')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nail/home/kwa/paasta/paasta_tools/spark_tools.py", line 61, in get_aws_credentials
    return _load_aws_credentials_from_yaml(aws_credentials_yaml)
  File "/nail/home/kwa/paasta/paasta_tools/spark_tools.py", line 33, in _load_aws_credentials_from_yaml
    with open(yaml_file_path, "r") as yaml_file:
FileNotFoundError: [Errno 2] No such file or directory: 'bla'
```

Can you make sure the user experience is still as good as it can be here?

Reply:
If a user provides typo in aws credential path name of paasta spark-run command. The tron action run would fail; If the typo happens to `aws_credentials_yaml` option, the `setup_tron_namespaces` batch run would fail. So I think user experience is as good, as long as `setup_tron_namespaces` batch run failures can be notified as efficiently as tron action run failures?

Reply:
s_t_n failures currently page the CI oncall.

If you can't figure out a great way to provide a super good user experience, can you figure out a way to meet it half-way to give the feedback to the CI oncall, making it very easy for them to understand what was typod? (instead of a stacktrace)

Reply:
Will do.

Reply:
Because it isn't easy for setup_tron_namespace to notify the user, and it just pages us instead, I think it would be better to catch any exceptions in loading the credentials in setup_tron_namespace and let the tron action fail later. I think my suggestion on the docstring will make it clear to the job owner what went wrong.

Also, can we let this default to `/etc/boto_cfg/{service}.yaml`? spark-run does that, and it would cover the large majority of cases. That way, there is less of an opportunity for typos.

Reply:
(lol I'm going to reset my credentials... Thanks @danielpops ...)